subsys/net/l2/wifi/wifi_mgmt.c

```diff
@@ -754,6 +754,29 @@ static int wifi_get_version(uint32_t mgmt_request, struct net_if *iface,
 
 NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_VERSION, wifi_get_version);
 
+#ifdef CONFIG_WIFI_NM_WPA_SUPPLICANT_WNM
+static int wifi_btm_query(uint32_t mgmt_request, struct net_if *iface,
+			   void *data, size_t len)
+{
+	const struct device *dev = net_if_get_device(iface);
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_api(iface);
+	uint8_t query_reason = *((uint8_t *)data);
+
+	if (wifi_mgmt_api == NULL || wifi_mgmt_api->btm_query == NULL) {
+		return -ENOTSUP;
+	}
+
+	if (query_reason >= WIFI_BTM_QUERY_REASON_UNSPECIFIED &&
```

Comment:
Should we limit to specific reasons?

```
	if (query_reason != WIFI_BTM_QUERY_REASON_UNSPECIFIED ||
		query_reason !=WIFI_BTM_QUERY_REASON_LOW_RSSI ||
		query_reason != WIFI_BTM_QUERY_REASON_LEAVING_ESS) {
		return -EINVAL;
	}
```

Reply:
This logic is to limit input parameter: 
if (query_reason >= WIFI_BTM_QUERY_REASON_UNSPECIFIED &&
	query_reason <= WIFI_BTM_QUERY_REASON_LEAVING_ESS)

Reply:
it's not really a big deal, but you are allowing the full range from `0-20` while valid reasons are only `0,16,20`

Reply:
We only enum 0,16,20 here, but input reason from 0-20 is correct and acceptable for supplicant